server/service/src/sync/translations/location_movement.rs

```diff
@@ -0,0 +1,168 @@
+    #[serde(serialize_with = "date_option_to_isostring")]
+    pub exit_date: Option<NaiveDate>,
+    #[serde(deserialize_with = "naive_time")]
+    pub exit_time: NaiveTime,
+}
+
+pub(crate) struct LocationMovementTranslation {}
+impl SyncTranslation for LocationMovementTranslation {
+    fn try_translate_pull_upsert(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        if !match_pull_table(sync_record) {
+            return Ok(None);
+        }
+
+        let LegacyLocationMovementRow {
+            id,
+            store_id,
+            stock_line_id,
+            location_id,
+            enter_date,
+            enter_time,
+            exit_date,
+            exit_time,
+        } = serde_json::from_str::<LegacyLocationMovementRow>(&sync_record.data)?;
+
+        let result = LocationMovementRow {
+            id,
+            store_id,
+            stock_line_id,
+            location_id,
+            enter_datetime: enter_date.map(|enter_date| NaiveDateTime::new(enter_date, enter_time)),
+            exit_datetime: exit_date.map(|exit_date| NaiveDateTime::new(exit_date, exit_time)),
+        };
+
+        Ok(Some(IntegrationRecords::from_upsert(
+            PullUpsertRecord::LocationMovement(result),
+        )))
+    }
+
+    fn try_translate_push_upsert(
+        &self,
+        connection: &StorageConnection,
+        changelog: &ChangelogRow,
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
+        if !match_push_table(changelog) {
+            return Ok(None);
+        }
+
+        let LocationMovementRow {
+            id,
+            store_id,
+            stock_line_id,
+            location_id,
+            enter_datetime,
+            exit_datetime,
+        } = LocationMovementRowRepository::new(connection)
+            .find_one_by_id(&changelog.record_id)?
+            .ok_or(anyhow::Error::msg(format!(
+                "Location movement row ({}) not found",
+                changelog.record_id
+            )))?;
+
+        let legacy_row = LegacyLocationMovementRow {
+            id: id.clone(),
+            store_id: store_id,
+            stock_line_id,
+            location_id,
+            enter_date: enter_datetime.map(|datetime| datetime.date()),
+            enter_time: enter_datetime
+                .map(|datetime| datetime.time())
+                .unwrap_or(NaiveTime::from_hms_opt(0, 0, 0).unwrap()),
+            exit_date: exit_datetime.map(|datetime| datetime.date()),
+            exit_time: exit_datetime
+                .map(|datetime| datetime.time())
+                .unwrap_or(NaiveTime::from_hms_opt(0, 0, 0).unwrap()),
+        };
+
+        Ok(Some(vec![RemoteSyncRecordV5::new_upsert(
+            changelog,
+            LEGACY_TABLE_NAME,
+            serde_json::to_value(&legacy_row)?,
+        )]))
+    }
+
+    fn try_translate_push_delete(
```

Comment:
technically I don't think we will be deleting any location_movements

Reply:
ok thanks, was briefly wondering about it, should I remove?

Reply:
Up to you really, it's not much code (although my preference would be to only have code that is used)